File: back/src/LaMaps.Desktop/Gpx/LaTrack.cs
using System.Collections.Generic;

namespace LaMaps.Desktop.Controllers
{
    public class LaTrack
    {
        public string Datum { get; }
        public string TrackWidth { get; }
        public string Color { get; }
        public string Description { get; }
        public IList<LaSegment> Segments { get; }

        public LaTrack(string datum, string trackWidth, string color, string description, IList<LaSegment> segments)
        {
            Datum = datum;
            TrackWidth = trackWidth;
            Color = color;
            Description = description;
            Segments = segments;
        }
    }
}

File: front/src/api/models/LaTrack.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { LaSegment } from './LaSegment';

export type LaTrack = {
    readonly datum?: string | null;
    readonly trackWidth?: string | null;
    readonly color?: string | null;
    readonly description?: string | null;
    readonly segments?: Array<LaSegment> | null;
}